src/main/java/org/aksw/iguana/cc/tasks/impl/StresstestResultProcessor.java

```diff
@@ -143,6 +143,14 @@ public void calculateAndSaveMetrics(Calendar start, Calendar end) {
             }
         }
 
+        // Create Query nodes with their respective queryIDs
+        for (String queryID : queryIDs) {
+            Resource queryRes = IRES.getResource(queryID);
+            m.add(queryRes, RDF.type, IONT.query);
+            m.add(queryRes, IPROP.fullID, ResourceFactory.createTypedLiteral(queryID));
+            m.add(queryRes, IPROP.id, ResourceFactory.createTypedLiteral(BigInteger.valueOf(Long.parseLong(queryID.split(":")[1]))));
```

Comment:
This looks dangerous. How does queryID look like and how is guaranteed that it contains only one colon?

It is a IRI, right? It might have a protocol and a port, both using a colon: `http://example.com:8000/`

Reply:
QueryID consists of the hashcode of its query handler (the list of queries it's a part of to be precise) and the id inside of that list in this format: <queryhandler_hash>:<id>